src/pages/AdvancedSearch/ResultsBlock/ResultsBlock.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Spinner,
+} from "@canonical/react-components";
+import { useState } from "react";
+
+import {
+  getCrossModelQueryLoaded,
+  getCrossModelQueryLoading,
+  getCrossModelQueryResults,
+} from "store/juju/selectors";
+import { useAppSelector } from "store/store";
+
+enum CodeSnippetView {
+  TREE = "tree",
+  JSON = "json",
+}
+
+const ResultsBlock = (): JSX.Element | null => {
+  const isCrossModelQueryLoading = useAppSelector(getCrossModelQueryLoading);
+  const isCrossModelQueryLoaded = useAppSelector(getCrossModelQueryLoaded);
+  const crossModelQueryResults = useAppSelector(getCrossModelQueryResults);
+  const [codeSnippetView, setCodeSnippetView] = useState<CodeSnippetView>(
+    CodeSnippetView.TREE
+  );
+  const codeSnippetContent = {
+    tree: "",
+    json: JSON.stringify(crossModelQueryResults, null, 2),
```

Comment:
Nice, I didn't realise stringify() could format as well. I've assigned you this task as well as it looks like it's been done: https://warthogs.atlassian.net/browse/WD-5817.